pkg/controller/monitor/monitor_controller_test.go

```diff
@@ -0,0 +1,134 @@
+
+package monitor
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/stretchr/testify/mock"
+
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/apis"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/render"
+)
+
+var _ = Describe("Monitor controller tests", func() {
+	var cli client.Client
+	var ctx context.Context
+	var mockStatus *status.MockStatus
+	var r ReconcileMonitor
+	var scheme *runtime.Scheme
+
+	BeforeEach(func() {
+		// The schema contains all objects that should be known to the fake client when the test runs.
+		scheme = runtime.NewScheme()
+		Expect(apis.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(appsv1.SchemeBuilder.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(rbacv1.SchemeBuilder.AddToScheme(scheme)).NotTo(HaveOccurred())
+
+		// Create a client that will have a crud interface of k8s objects.
+		ctx = context.Background()
+		cli = fake.NewClientBuilder().WithScheme(scheme).Build()
+
+		// Create an object we can use throughout the test to do the monitor reconcile loops.
+		mockStatus = &status.MockStatus{}
+		mockStatus.On("AddCronJobs", mock.Anything)
+		mockStatus.On("AddDaemonsets", mock.Anything)
+		mockStatus.On("AddDeployments", mock.Anything).Return()
+		mockStatus.On("AddStatefulSets", mock.Anything)
+		mockStatus.On("ClearDegraded")
+		mockStatus.On("IsAvailable").Return(true)
+		mockStatus.On("OnCRFound").Return()
+		mockStatus.On("ReadyToMonitor")
+
+		// Create an object we can use throughout the test to do the monitor reconcile loops.
+		r = ReconcileMonitor{
+			client:   cli,
+			scheme:   scheme,
+			provider: operatorv1.ProviderNone,
+			status:   mockStatus,
+		}
+
+		// We start off with a 'standard' installation, with nothing special
+		Expect(cli.Create(ctx, &operatorv1.Installation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "default",
+			},
+			Status: operatorv1.InstallationStatus{
+				Variant:  operatorv1.TigeraSecureEnterprise,
+				Computed: &operatorv1.InstallationSpec{},
+			},
+			Spec: operatorv1.InstallationSpec{
+				Variant:  operatorv1.TigeraSecureEnterprise,
+				Registry: "some.registry.org/",
+			},
+		})).To(BeNil())
+
+		// Apply the Monitor CR to the fake cluster.
+		Expect(cli.Create(ctx, &operatorv1.Monitor{
+			TypeMeta:   metav1.TypeMeta{Kind: "Monitor", APIVersion: "operator.tigera.io/v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+		})).NotTo(HaveOccurred())
+	})
+
+	Context("image reconciliation", func() {
```

Comment:
nit: The context does not really seem to be about image reconciliation, it seems to be about the reconciliation of the controller.